fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/expressions/SelectExpression.java

```diff
@@ -210,25 +218,26 @@ public int hashCodeWithoutChildren() {
             return ImmutableList.of();
         }
 
-        // loop through all for each quantifiers on the other side to ensure that they are all matched
+        // Loop through all for each quantifiers on the other side to ensure that they are all matched.
+        // If any are not matched we cannot establish a match at all.
         final boolean allOtherForEachQuantifiersMatched =
                 otherSelectExpression.getQuantifiers()
                         .stream()
                         .filter(quantifier -> quantifier instanceof Quantifier.ForEach)
                         .allMatch(quantifier -> aliasMap.containsTarget(quantifier.getAlias()));
 
+        // TODO this is not really needed if we assign a property to the quantifier that allows us to reason about the
+        // TODO the "default on empty" property as in does this quantifier flow a scalar result such as an "empty" value,
+        // TODO a real value, or even null if the underlying graph evaluates to empty. The presence of such a property
+        // TODO would help us here to make sure the additional non-matched quantifier is not eliminating records.
```

Comment:
You don't really need to repeat `TODO`; just indenting make a multi-line todo in the IDE.

Reply:
TIL